feat: navigator on main page

loopnyapy opened the pull request:
feat: four squares page
feat: navigation on pages

loopnyapy committed:
feat: navigator on main page
feat: four squares page
feat: navigation on pages

dkrutskikh commented on lib/presentation/widgets/custom_appbar.dart:
```diff
@@ -2,22 +2,28 @@ import 'package:flutter/material.dart';
+              onPressed: () => Navigator.pop(context),
+              icon: const Icon(Icons.arrow_back),
+            )
+          : Container(),
```
@loopnyapy зачем тут контейнер, вроде как можно передать null

loopnyapy replied:
нулл убирает это пробел на месте иконки
![image](https://user-images.githubusercontent.com/62483370/148639323-03652261-93a2-4bb1-a7df-148639ba820c.png)
![image](https://user-images.githubusercontent.com/62483370/148639338-da58a9b7-2dd1-46d6-9f24-dcef78988e2c.png)

dkrutskikh commented on lib/presentation/main/widgets/menu_item.dart:
```diff
@@ -1,28 +1,34 @@
+  const MenuItem({Key? key, required this.title, required this.onTap}) : super(key: key);
 
   final String title;
+  final Function onTap;
```
@loopnyapy сейчас судя по твоему коду onTap принимает dynamic и возвращает dynamic. Давай типизируем эти места

dkrutskikh commented on lib/presentation/main/widgets/menu_item.dart:
```diff
@@ -1,28 +1,34 @@
-    return Container(
-      padding: const EdgeInsets.symmetric(vertical: 16.0, horizontal: 20.0),
-      decoration: BoxDecoration(
+    return InkWell(
```
@loopnyapy давай переедем на кнопку

dkrutskikh commented on lib/presentation/four_squares/widgets/square.dart:
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter/material.dart';
+
+class Square extends StatefulWidget {
```
@loopnyapy два вопроса:

1) почему StatefulWidget тк в state только функция build?
2) почему квадрат безразмерный?

loopnyapy replied:
1. Да, исправил, вообще этот виджет даже не использовал в этом пр, поэтому не заметил.
2. Чтобы они резиновые были, гридом все делится на квадраты, а их размеры автоматом растягиваются на весь экран

dkrutskikh commented on lib/presentation/four_squares/four_squares_page.dart:
```diff
@@ -0,0 +1,48 @@
+        children: List.generate(
+          4,
+          (_) => Container(
+            color: generateColor(),
```
@loopnyapy получается что каждую перерисовку виджета у тебя будет новый цвет, это точно то что ты хотел получить?

loopnyapy committed:
fix: генерация цветов только по нажатию
fix: floatingActionButton квадратная
fix: MenuItem.inkWell -> ElevatedButton
fix: Square отдельный виджет вместо контейнера

dkrutskikh commented on lib/presentation/four_squares/four_squares_page.dart:
```diff
@@ -0,0 +1,75 @@
+// ignore_for_file: prefer_const_constructors
```
@loopnyapy не надо игнорировать это предупреждение, очень много оптимизаций основаны на константности данных

loopnyapy replied:
ой, это я случайно :0 исправлю

dkrutskikh commented on lib/presentation/four_squares/four_squares_page.dart:
```diff
@@ -0,0 +1,75 @@
+
+class _FourSquarePageState extends State<FourSquarePage> {
+  static const squareListLength = 4;
+  List<Square> currentSquareList = [];
```
@loopnyapy не стоит класть в состояние виджеты, их надо создавать в build функции

loopnyapy committed:
fix: FourSquarePage widgets in state

loopnyapy commented:
а если цвета хранить и по ним генерировать?
